x-pack/filebeat/input/httpjson/value_tpl.go

```diff
@@ -295,6 +303,186 @@ func div(a, b int64) int64 {
 	return a / b
 }
 
+// minimum returns the minimum of a and an arbitrary number of values.
+func minimum(a interface{}, nums ...interface{}) (interface{}, error) {
+	var min interface{}
+	var err error
+
+	min = math.MaxInt
+	nums = append(nums, a)
+	if len(nums) == 1 {
+		return nums[0], nil
+	}
+	for _, num := range nums {
+		min, err = findMin(min, num)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return min, nil
+}
+
+// findMin returns the minimum of two values.
+func findMin(b, a interface{}) (interface{}, error) {
```

Comment:
The mixing of float and integer types in these functions makes me feel extremely uneasy. I would much rather see integer only and float only (why do we need float coverage here?) and even then, I'm still not wildly comfortable with the loss of safety that comes from comparing values that have different sign ranges.

Reply:
@efd6 Initially I was going for an int only approach but then I saw the consul code which had generic min/ max funcs that covered all cases and thought I could make it as generic as possible which would be better than having minInt , minFloat funcs

Reply:
also the generic math.Min() and math.Max() funcs in go accept float64s and return a float64 which would provides the highest base range for values. So in order of keeping the same flexibility won't the generic nature of this method be for the better for the user who would prefer a single min/max func for all required purposes ?

Reply:
`math.Min` and `math.Max` aren't generic, they are `float64`.

If we had a `min[T number](head T, tail ..T) T` and `min[T number](head T, tail ..T) T` where `type number interface { ~int | ~int8 | ~int16 | ~int32 | ~int64 | ~uint | ~uint8 | ~uint16 | ~uint32 | ~uint64 | ~uintptr | ~float32 | ~float64 }`, I'd be fine with that, but it cannot be used to satisfy the type system when constructing the template's func map.

Reply:
To be clear, I think I would be OK with a generalised loosely typed collection comparison function if it were more careful about ranges. This would require that values that are being compared MUST be comparable and error otherwise. So int values with an absolute value >1e52 would error when compared with a float64 and negative integer values would error when compared with an unsigned integer. The code would become significantly more complex with these additions.

Reply:
Yea, it's not worth doing that atm, I will segregate these into separate funcs so we avoid all these range and sign issues.